### PR TITLE
DNS: add canary

### PR DESCRIPTION
**Pull request by thockin**
Add canary-push capability and a glimmer of a test.  Need help with the test - anyone have time?

@kubernetes/k8s-infra-team

### CODE REVIEW

**Comment by thockin** [comment]
Waiting for approval.  @hh I think you said "merge and iterate" as an approach, so it's in your court :)

File: dns/test.sh [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+    echo "  got:"
+    echo "$R"
+    echo "--------------------"
+    exit 2
+fi
+
+echo "PASS"
+
+exit 0
+### Input data for k8s.io below...
```
Comment (hh) [inline dns/test.sh]: @thockin I'm unsure how this test data below is accessed.

Comment (hh) [reply dns/test.sh]: I'm leaning towards loading the zone-configs directly for the verification.

**Comment by hh** [comment]
I'm thinking a python script that iterates over zone-configs and verifies all entries with http://www.dnspython.org/.

**Comment by thockin** [comment]
@hh Python or Go is fine to me - the Go test doesn't seem that hard to write, but my Go is better than my Python.  Can we merge this and get the test in as followup?

**Comment by thockin** [comment]
Oh, this again.

@hh approved

**Pull request merged**